### ISSUE TITLE
DataContractSerializer causes InvalidDataException

### ISSUE THREAD

**Issue by mejje** [MiloszKrajewski/K4os.Compression.LZ4#22]
Hello, I'm having problems using this library with a DataContractSerializer. This example throws System.IO.InvalidDataException: LZ4 frame magic number expected. Can't figure it out, and it works fine copying to an intermediate MemoryStream for some reason (but of course I want to avoid that).
```
at K4os.Compression.LZ4.Streams.LZ4DecoderStream.ReadFrame()
   at K4os.Compression.LZ4.Streams.LZ4DecoderStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.BufferedStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.Xml.EncodingStreamWrapper.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Xml.XmlBufferReader.TryEnsureBytes(Int32 count)
   at System.Xml.XmlUTF8TextReader.BufferElement()
   at System.Xml.XmlUTF8TextReader.ReadStartElement()
   at System.Xml.XmlUTF8TextReader.Read()
   at System.Xml.XmlBaseReader.IsStartElement()
   at System.Xml.XmlBaseReader.IsStartElement(XmlDictionaryString localName, XmlDictionaryString namespaceUri)
   at System.Runtime.Serialization.XmlReaderDelegator.IsStartElement(XmlDictionaryString localname, XmlDictionaryString ns)
   at System.Runtime.Serialization.XmlObjectSerializer.IsRootElement(XmlReaderDelegator reader, DataContract contract, XmlDictionaryString name, XmlDictionaryString ns)
   at System.Runtime.Serialization.DataContractSerializer.InternalIsStartObject(XmlReaderDelegator reader)
   at System.Runtime.Serialization.DataContractSerializer.InternalReadObject(XmlReaderDelegator xmlReader, Boolean verifyObjectName, DataContractResolver dataContractResolver)
   at System.Runtime.Serialization.XmlObjectSerializer.ReadObjectHandleExceptions(XmlReaderDelegator reader, Boolean verifyObjectName, DataContractResolver dataContractResolver)
   at System.Runtime.Serialization.XmlObjectSerializer.ReadObject(XmlDictionaryReader reader)
   at System.Runtime.Serialization.XmlObjectSerializer.ReadObject(Stream stream)
   at CitybreakOnlineWS.Tests.App.Utils.LZ4Tests.LibEncodeDecode() in LZ4Tests.cs:line 38
```

```
    [TestClass]
    public class LZ4Tests
    {
        public class TestClass
        {
            public string Test1 { get; set; }
        }

        [TestMethod]
        public void LibEncodeDecode()
        {
            var obj = new TestClass { Test1 = "1" };
            var serializer = new DataContractSerializer(typeof(TestClass));

            byte[] bytes;
            using (var ms = new MemoryStream())
            {
                using (var compressionStream = LZ4Stream.Encode(ms))
                {
                    serializer.WriteObject(compressionStream, obj);
                }
                bytes = ms.ToArray();
            }

            using (var ms = new MemoryStream(bytes))
            {
                using (var decompressionStream = LZ4Stream.Decode(ms))
                {
                    var o = serializer.ReadObject(decompressionStream) as TestClass;

                    Assert.AreEqual(obj.Test1, o.Test1);
                }
            }
        }
    }
```

**Comment by MiloszKrajewski** [comment]
Caused by #18. Seems like it wasn't a good idea.

**Comment by MiloszKrajewski** [comment]
Fixed in 1.1.7

**Comment by mejje** [comment]
Thank you, it works with the default datacontractserializer (text), but when using a binary reader/writer it causes a SerializationException:
```
System.Runtime.Serialization.SerializationException: There was an error deserializing the object of type LZ4Tests+TestClass. The input source is not correctly formatted. ---> System.Xml.XmlException: The input source is not correctly formatted.
```

Verified test with GZipStream.

```
        [TestMethod]
        public void LibEncodeDecode_Binary()
        {
            var obj = new TestClass { Test1 = "1" };
            var serializer = new DataContractSerializer(typeof(TestClass));

            byte[] bytes;
            using (var ms = new MemoryStream())
            {
                using (var compressionStream = LZ4Stream.Encode(ms))
                //using (var compressionStream = new System.IO.Compression.GZipStream(ms, System.IO.Compression.CompressionMode.Compress))
                {
                    using (var writer = System.Xml.XmlDictionaryWriter.CreateBinaryWriter(compressionStream))
                    {
                        serializer.WriteObject(writer, obj);
                    }
                }
                bytes = ms.ToArray();
            }

            using (var ms = new MemoryStream(bytes))
            {
                using (var decompressionStream = LZ4Stream.Decode(ms))
                //using (var decompressionStream = new System.IO.Compression.GZipStream(ms, System.IO.Compression.CompressionMode.Decompress))
                {
                    using (var reader = System.Xml.XmlDictionaryReader.CreateBinaryReader(decompressionStream, System.Xml.XmlDictionaryReaderQuotas.Max))
                    {
                        var o = serializer.ReadObject(reader) as TestClass;

                        Assert.AreEqual(obj.Test1, o.Test1);
                    }
                }
            }
        }
```

**Comment by MiloszKrajewski** [comment]
...and fixed again, in 1.1.9 this time
